sklearn/cross_decomposition/pls_.py

```diff
@@ -820,7 +820,14 @@ def fit(self, X, Y):
         return self
 
     def transform(self, X, Y=None):
-        """Apply the dimension reduction learned on the train data."""
+        """Apply the dimension reduction learned on the train data.
+
+        Parameters
+        ----------
+        X : array-like of predictors, shape = [n_samples, p]
+            Training vectors, where n_samples in the number of samples and
+            p is the number of predictors.
+        """
```

Comment:
Ah sorry this is an unrelated change. I'll revert it. I want to leave PLS issue for another PR ( #9160 ) and the documentation to yet another. (https://github.com/scikit-learn/scikit-learn/pull/9023)